x-pack/plugin/ent-search/qa/full-cluster-restart/javaRestTest/java/org/elasticsearch/entsearch/FullClusterRestartIT.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+package org.elasticsearch.entsearch;
+
+import com.carrotsearch.randomizedtesting.annotations.Name;
+
+import org.elasticsearch.client.Request;
+import org.elasticsearch.client.Response;
+import org.elasticsearch.test.cluster.ElasticsearchCluster;
+import org.elasticsearch.test.cluster.local.distribution.DistributionType;
+import org.elasticsearch.test.rest.ObjectPath;
+import org.elasticsearch.upgrades.FullClusterRestartUpgradeStatus;
+import org.elasticsearch.upgrades.ParameterizedFullClusterRestartTestCase;
+import org.junit.ClassRule;
+
+import java.io.IOException;
+import java.util.List;
+
+public class FullClusterRestartIT extends ParameterizedFullClusterRestartTestCase {
```

Comment:
I would prefer a more explicit test naming:
```suggestion
public class DataStreamLifecycleMigrationIT extends ParameterizedFullClusterRestartTestCase {
```

Reply:
I disagree, this is a very lightweight test right now and there's precedent for `FullClusterRestartIT` in other modules. I think it lowers the barrier of entry to adding more tests to keep it as-is until we deem otherwise.

Reply:
I agree with @afoucret as this currently holds tests for analytics. But I think @kderusso is aiming for a pattern that is present in other areas of the codebase - doing a specific test for cluster restarting, where we can add new tests for migrations. 

I'd stick with a common codebase pattern even if my first gut feeling is to create a more specific, per-feature level test class.

Reply:
Both patterns exists into the code (eg `org.elasticsearch.xpack.restart.WatcherMappingUpdateIT`. 
I prefer using one class per test with an explicit naming.

Reply:
I'm being worn down. 🙂 If you both feel strongly that I change it I will, otherwise I'll keep it as is.